lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -127,11 +127,31 @@ def get_current_span() -> Union[Span, None]:
     return cast(Span, span)
 
 
+def _get_tracer_from_context(ctx: Context) -> Optional[ContextVar]:
+    try:
+        tracers = [v for v in ctx if v is not None and v.name == "tracer"]
+        if tracers:
+            return tracers[0]
+    except StopIteration:
+        return
```

Comment:
Does `ctx` raise a `StopIteration` exception? 🤔

Reply:
Right, that was a leftover from previous approach. Removed the catch now.